url/toascii.window.js

```diff
@@ -0,0 +1,65 @@
+async_test(t => {
+  const request = new XMLHttpRequest()
+  request.open("GET", "toascii.json")
+  request.send()
+  request.responseType = "json"
+  request.onload = t.step_func_done(() => {
+    runTests(request.response)
+  })
+}, "Loading data…")
+
+function makeURL(type, input) {
+  input = "https://" + input + "/x"
+  if(type === "url") {
+    return new URL(input)
+  } else {
+    const url = document.createElement(type)
+    url.href = input
+    return url
+  }
+}
+
+function runTests(tests) {
+  for(var i = 0, l = tests.length; i < l; i++) {
+    let hostTest = tests[i]
+    if (typeof hostTest === "string") {
+      continue // skip comments
+    }
+    if(hostTest.output === undefined) {
```

Comment:
I would prefer that this kind of normalization be done ahead of time, so that there is less logic in the tests. I will push a commit to fix.